modules/mono/glue/GodotSharp/GodotSharp/Core/Array.cs

```diff
@@ -273,20 +478,82 @@ public void Add(Variant item)
         }
 
         /// <summary>
-        /// Checks if this <see cref="Array"/> contains the given item.
+        /// Adds the elements of the specified collection to the end of this <see cref="Array"/>.
+        /// </summary>
+        /// <param name="collection">Collection of <see cref="Variant"/> items to add.</param>
+        /// <exception cref="ArgumentNullException">
+        /// The <paramref name="collection"/> is <see langword="null"/>.
+        /// </exception>
+        public void AddRange(IEnumerable<Variant> collection)
+        {
+            if (collection == null)
+                throw new ArgumentNullException(nameof(collection), "Value cannot be null.");
+
+            // If the collection is another Godot Array, we can add the items
+            // with a single interop call.
+            if (collection is Array array)
+            {
+                var self = (godot_array)NativeValue;
+                var collectionNative = (godot_array)array.NativeValue;
+                _ = NativeFuncs.godotsharp_array_add_range(ref self, collectionNative);
+                return;
+            }
+            if (collection is Array<Variant> typedArray)
```

Comment:
What I really wanted here was:

```csharp
if (collection is Array<> typedArray)
```

Since none of the code inside the `if` block cares about the value of the generic parameter `T`, but you can't do that with C# today.

May not be worth keeping this `if` block for the untyped Array method, since I guess the chances of users using `Array<Variant>` rather than an untyped `Array` would be low.